### DISCUSSION TITLE
Include or read "meta" data about images from .mind file?

### DISCUSSION THREAD

**Discussion by uptownben** [hiukim/mind-ar-js#249] [Q&A]
When a `.mind` file is created it is created with images in the order they are uploaded to the .mind "compiler". I am looking for a way to know more about the contents of the `.mind` file so that my web logic can be more dynamic. If I receive a `.mind` file that I did not create, I have no way of knowing how many targets it contains, also no way to identify each target except for its sequence number (index) within the `.mind` file.  Looking over the `Event` object that is emitted on `targetFound`, but it does not contain data about the target in the `.mind` file.  Is there currently any way to know anything about the target that was "found" besides its index within the `.mind` file? If not, maybe an optional "flag" can be added to the compiler code that, when "true" will tell the compiler to generate an accompanying JSON file with filenames, indexes and some option string (meta)data to match the items in the `.mind` file. This way the `.mind` file and the JSON file can be distributed together and loaded dynamically and used to know more about the targets that are found. What are your thoughts about this? @hiukim ?

**Comment by hiukim** [comment]
It depends on what you mean by "identify".  filename is currently not stored in the .mind file.  

I'm curious about your use case though?  I mean, if your users compile the files to you without any communication with you, then the `filenames` still aren't reliable, since they could be anything you don't know.  If your users can communicate with you, they can tell you which index corresponds to which targets.
